### PR TITLE
Gh-407: Add configuration page to Administration guide

### LINKED ISSUE #407: Document configuration settings for Gaffer instance

### PR DESCRIPTION
**Pull request by rj77259**

### CODE REVIEW

File: docs/administration-guide/gaffer-config/config.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    ]
+}
+```
+
+The two additional operations already exist in Gaffer (in the code base:
+[ImportFromLocalFile](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ImportFromLocalFile.java)
+and
+[ExportToLocalFile](https://github.com/gchq/Gaffer/blob/develop/core/operation/src/main/java/uk/gov/gchq/gaffer/operation/impl/export/localfile/ExportToLocalFile.java)),
+what this file is doing is essentially activating them and setting the handler class for them. The
+`ImportFromLocalFile` usage is demonstrated in the [using the API](./using-the-api.md) section to
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]: CI has flagged this link as incorrect

File: docs/administration-guide/gaffer-config/config.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+gaffer.store.operation.declarations=/gaffer/store/operationsDeclarations.json
+```
+
+### Operations Declarations
+
+The operation declarations file is a way of enabling additional operations in Gaffer. By default
+there are some built in operations already available (the rest API has a get all operations request
+to see a list), but its likely you might want to enable others or add your own custom ones. As the
+example will load its data from a local CSV file we can activate a couple of additional operations
+using the following file.
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]:
```suggestion
The operation declarations file is a way of enabling additional operations in Gaffer. By default
most built-in operations are already available (use `GetAllOperations` or the equivalent REST
endpoint to see a list), but it's possible you'll want to enable others or add your own custom ones.
The example loads its data from a local CSV file, we can activate a couple of additional operations
for this using the following file.
```

File: docs/administration-guide/gaffer-config/config.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+```json title="graphConfig.json"
+{
+    "graphId": "ExampleGraph",
+    "description": "An example graph"
+}
+```
+
+### Store Properties
+
+The store properties file is used to configure how Gaffer will store its data. There are a few
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]:
```suggestion
The store properties file is used to configure the Gaffer store for a graph. There are a few
```

File: docs/administration-guide/gaffer-config/config.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+different stores available for Gaffer, these are explained in more detail in the [reference
+documentation](../../administration-guide/gaffer-stores/store-guide.md), but by default you must provide a store
+class and a store properties class. For this example we are using an Accumulo store as it is
+recommended for efficient storage and retrieval of large data volumes. It's set up requires a few
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]:
```suggestion
recommended for efficient storage and retrieval of large data volumes. Its configuration requires a few
```

File: docs/administration-guide/gaffer-config/config.md [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+gaffer.schemas=/gaffer/schema
+gaffer.storeProperties=/gaffer/store/store.properties
+gaffer.graph.config=/gaffer/graph/graphConfig.json
+```
+
+### Graph Configuration
+
+The graph configuration file is a JSON file that configures few bits of the Gaffer graph. Primarily
+it is used to set the name and description along with any additional hooks to run before an operation
+chain e.g. to impose limits on max results etc. For the example as, it is a very basic graph we just
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]:
```suggestion
chain e.g. to impose limits on max results etc. For the example, as it is a very basic graph we just
```

File: docs/administration-guide/gaffer-config/config.md [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+[here](https://github.com/gchq/gaffer-docker/tree/develop/docker/accumulo/conf-2.0.1) and
+[here](https://github.com/gchq/gaffer-docker/tree/develop/docker/hdfs/conf). The configuration of
+these are out of scope of this example but are covered in other sections of the documentation. The
+main focus of this guide will be on the configuration files under the `config/gaffer/` directory.
+
+## Configuration Files
+
+### Application Properties
+
+This is probably the simplest configuration file in the Gaffer deployment. In general it borrows a
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md:57]: I'd say that the `graphConfig.json` is simpler 😄

File: docs/administration-guide/gaffer-config/config.md [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+concept from [Spring Boot](https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html)
+to allow changing the context root and any properties related to Gaffer. In the example that follows
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]: I'm not aware of this changing the context root for Gaffer.

File: docs/administration-guide/gaffer-config/config.md [hunk 8]
```diff
@@ -0,0 +1,150 @@
+## How to configure a Gaffer instance
+
+First you must set up the files and directories you will need for the instance. As it stands there
+are a couple of different ways to run a Gaffer project this example will use a logical structure
+that suites a stand alone deployment consisting of the following file structure:
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]: Looks like this is using Docker Compose so may want to mention that here.

File: docs/administration-guide/gaffer-config/config.md [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+    │           └── store.properties            #(7)!
+    │   
+    └── docker-compose.yaml #(8)!
+    ```
+
+    1. Properties file that generally sets the file locations of other Gaffer
+        configs e.g. schemas (note these are the absolute paths inside the
+        container).
+    2. Any data files, e.g. CSV, to be made available to the Gaffer container.
+    3. The main graph config file to set various properties of the overall graph.
```
Comment (GCHQDeveloper314) [inline docs/administration-guide/gaffer-config/config.md]:
```suggestion
    3. The graph config file to set id, description and other values for the graph.
```

File: docs/administration-guide/gaffer-config/config.md [hunk 10]
```diff
@@ -0,0 +1,150 @@
...
+    4. This file holds the schema outlining the elements in the graph, e.g. the
+    nodes (aka entities) and edges.
```
Comment (t92549) [inline docs/administration-guide/gaffer-config/config.md]: The Gaffer terminology is Entities and Edges, so I think if you mention node it should be in brackets

File: docs/administration-guide/gaffer-config/config.md [hunk 11]
```diff
@@ -0,0 +1,150 @@
...
+    5. This file defines the different data types in the graph and how they are
```
Comment (t92549) [inline docs/administration-guide/gaffer-config/config.md:37]: These could probably point to schema documentation

Comment (rj77259) [reply docs/administration-guide/gaffer-config/config.md:37]: done

File: docs/administration-guide/gaffer-config/config.md (hunk 10 above)
Comment (rj77259) [reply docs/administration-guide/gaffer-config/config.md]: done

File: docs/administration-guide/gaffer-config/config.md (hunk 8 above)
Comment (rj77259) [reply docs/administration-guide/gaffer-config/config.md]: done

File: docs/administration-guide/gaffer-config/config.md (hunk 7 above)
Comment (rj77259) [reply docs/administration-guide/gaffer-config/config.md]: removed

File: docs/administration-guide/gaffer-config/config.md (hunk 1 above)
Comment (rj77259) [reply docs/administration-guide/gaffer-config/config.md]: fixed

**Review by t92549 — APPROVED** [review]

**Review by GCHQDeveloper314 — APPROVED** [review]

**Pull request merged**